dnd5esheets/front/src/components/CharacterList.tsx

```diff
@@ -1,21 +1,36 @@
 import { A } from '@solidjs/router'
-import { createSignal } from 'solid-js'
-import { css } from 'solid-styled'
 
 import { ListCharacterSchema } from '~/5esheet-client'
 
 export default function CharacterList({
   characters,
+  addCharacter,
+  state,
 }: {
   characters: ListCharacterSchema[]
+  addCharacter: (name: string) => void
 }) {
+  // not ideal to use this let and ref to access the input, but we can change that later
+  let input: HTMLInputElement
```

Comment:
Where does the value come from?

Reply:
`input` is a variable that is used with `ref={}` below to bind the HTML element to a Javascript variable: https://www.solidjs.com/tutorial/bindings_refs
Basically, it runs in this order (even if `ref` is initialized below its reference):
``` typescript
let input: HTMLInputElement
ref = <input></input> // actually <input ref={input}></input>
// ...
addCharacter(input.value)
```